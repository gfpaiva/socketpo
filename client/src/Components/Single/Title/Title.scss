@import '../../../Scss/root';

.single {
	&__title {
		display: flex;
		justify-content: space-between;
		padding: 1em 0;
	}

	&__status {
		background: $primary-color;
		border-radius: 50%;
		display: inline-block;
		height: 10px;
		margin-left: .5rem;
		width: 10px;

		&--stand-by { background: $status-warn; }
		&--in-progress { background: $status-ok; }
		&--finished { background: $status-notice; }
	}
}
