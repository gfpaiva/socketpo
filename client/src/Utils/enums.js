import React from 'react';
import Avatar from 'avataaars';

import {
	Rock,
	Paper,
	Scissors,
} from '../Components/Icons/Icons';

const statusValues = {
	0: 'Stand By',
	1: 'In progress',
	2: 'Finished'
};

const playValues = {
	0: 'Miss Play',
	1: 'Rock',
	2: 'Paper',
	3: 'Scissors'
};

const playIcons = {
	1: <Rock />,
	2: <Paper />,
	3: <Scissors />
}

const avatarValues = {
	1: <Avatar
		topType='LongHairMiaWallace'
		accessoriesType='Prescription02'
		hairColor='BrownDark'
		facialHairType='Blank'
		clotheType='Hoodie'
		clotheColor='PastelBlue'
		eyeType='Happy'
		eyebrowType='Default'
		mouthType='Smile'
		skinColor='Light'
	/>,
	2: <Avatar
		topType='LongHairBigHair'
		accessoriesType='Blank'
		hairColor='Blonde'
		facialHairType='Blank'
		clotheType='ShirtVNeck'
		clotheColor='PastelYellow'
		eyeType='Wink'
		eyebrowType='RaisedExcitedNatural'
		mouthType='Serious'
		skinColor='DarkBrown'
	/>,
	3: <Avatar
		topType='ShortHairShortFlat'
		accessoriesType='Prescription01'
		hairColor='Blonde'
		facialHairType='Blank'
		clotheType='BlazerSweater'
		eyeType='Default'
		eyebrowType='DefaultNatural'
		mouthType='Serious'
		skinColor='Pale'
	/>,
	4: <Avatar
		topType='ShortHairShortWaved'
		accessoriesType='Blank'
		hairColor='Auburn'
		facialHairType='BeardLight'
		facialHairColor='Red'
		clotheType='BlazerShirt'
		clotheColor='Gray01'
		eyeType='WinkWacky'
		eyebrowType='UnibrowNatural'
		mouthType='Grimace'
		skinColor='DarkBrown'
	/>,
};

export const parseStatus = statusNum => statusValues[statusNum];
export const parsePlay = playNum => playValues[playNum];
export const parsePlayIcons = playNum => playIcons[playNum];
export const parseAvatar = avatarNum => avatarValues[avatarNum];
