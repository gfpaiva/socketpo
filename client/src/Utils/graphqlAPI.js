import gql from 'graphql-tag';

export const totalGameFields = `
name
hash
status
players {
	id
	name
	avatar
	ready
	roundsWin
}
results {
	message
	matchWinner {
		id
		name
		avatar
	}
	rounds {
		isDraw
		finished
		plays {
			player {
				id
				name
			}
			play
		}
		winner {
			id
			name
			avatar
		}
	}
}`;

export const getGame = gql`
	query getGame($hash: String!) {
		GameByHash(hash: $hash) {
			${totalGameFields}
		}
	}
`;

export const play = gql`
	mutation play($hash: String!, $player: PlayerInput!, $play: Int! ) {
		play(hash: $hash, player: $player, play: $play ) {
			hash
			status
		}
	}
`;

export const gameSub = gql`
	subscription gameSub($hash: String!) {
		gameSubscription(hash: $hash) {
			game {
				${totalGameFields}
			}
			message
			hash
			player
		}
	}
`;
