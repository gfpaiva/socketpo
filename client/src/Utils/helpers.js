export const sanitize = str => (str.toLowerCase()
.replace(/[àáâãäå]/ig, 'a')
.replace(/[éèëê]/ig, 'e')
.replace(/[íìïî]/ig, 'i')
.replace(/[óòõöô]/ig, 'o')
.replace(/[uúùüû]/ig, 'u')
.replace(/[ç]/ig, 'c')
.replace(/^[0-9]/ig, '')
.replace(/\s/ig, '-'));
