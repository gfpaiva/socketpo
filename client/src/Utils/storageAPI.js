export const setObject = (key, object) => localStorage.setItem(key, JSON.stringify(object));
export const getObject = key => JSON.parse(localStorage.getItem(key));
