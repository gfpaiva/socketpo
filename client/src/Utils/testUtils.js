export const fakePlayer = (rewrite) => ({
	__typename: 'Player',
	avatar: 3,
	id: 'player01',
	name: 'Player 01',
	ready: false,
	roundsWin: 0,
	...rewrite
});

export const fakeGame = (fullPlayers = true, matchWinner = null, rounds = [], rewrite) => {

	const players = fullPlayers ? [fakePlayer(), fakePlayer({ id: 'player02', name: 'Player 02' })] : [fakePlayer()]

	return {
		__typename: 'Game',
		id: 'game01',
		hash: 'game01',
		name: 'Game 01',
		players,
		results: {
			__typename: "Result",
			matchWinner,
			message: null,
			rounds
		},
		status: 0,
		...rewrite
	};
};

export const fakeRound = (rewrite) => ({
	__typename: 'Round',
	isDraw: false,
	finished: true,
	plays: [
		fakePlay({ play: 1 }),
		fakePlay({ player: fakePlayer({ id: 'player02', name: 'Player 02' }), play: 3 })
	],
	winner: fakePlayer(),
	...rewrite
});

export const fakePlay = (rewrite) => ({
	__typename: 'Play',
	player: fakePlayer(),
	play: 0,
	...rewrite
});
