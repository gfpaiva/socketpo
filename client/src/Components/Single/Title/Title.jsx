import React from 'react';
import PropTypes from 'prop-types';

import { Query } from 'react-apollo';
import { withRouter } from 'react-router-dom';

import { parseStatus } from '../../../Utils/enums';
import { sanitize } from '../../../Utils/helpers';
import { getGame } from '../../../Utils/graphqlAPI';

import './Title.scss';

const Title = ( { match } ) => {

	return (
		<Query
			query={getGame}
			variables={{
				hash: match.params.hash
			}}
		>
			{({ data, loading, error }) => {
				if(loading || error) return null;

				const game = data.GameByHash;

				return (
					<div className="single__title">
						<p>{game.name}</p>
						<p>
							<strong>Status: </strong>{parseStatus(game.status)}
							<span className={`single__status single__status--${sanitize(parseStatus(game.status))}`}></span>
						</p>
					</div>
				);
			}}
		</Query>
	);
};

Title.propTypes = {
	match: PropTypes.object.isRequired
}

export default withRouter(Title);
