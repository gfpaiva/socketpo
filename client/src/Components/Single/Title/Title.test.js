import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import Title from './Title';
import { getGame } from '../../../Utils/graphqlAPI';
import { fakeGame } from '../../../Utils/testUtils';

const game = fakeGame();
const mocks = [
	{
		request: { query: getGame, variables: { hash: game.hash } },
		result: {
			data: {
				GameByHash: {
					...game
				}
			}
		},
	},
];

describe('<Title />', () => {

	it('should mount properly', async () => {
		const wrapper = mount(
			<MockedProvider mocks={mocks}>
				<Title.WrappedComponent match={{params: {hash: game.hash}}} />
			</MockedProvider>
		);

		await wait();
		wrapper.update();

		expect(wrapper.find('.single__title')).toMatchSnapshot();
	});
});
