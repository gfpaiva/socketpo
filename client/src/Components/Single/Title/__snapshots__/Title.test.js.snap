// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should mount properly 1`] = `
<div
  className="single__title"
>
  <p>
    Game 01
  </p>
  <p>
    <strong>
      Status: 
    </strong>
    Stand By
    <span
      className="single__status single__status--stand-by"
    />
  </p>
</div>
`;
